### PR TITLE
feat: new plugin configuration

### PR DESCRIPTION
**Pull request by zhangHongEn**

### CODE REVIEW

**Comment by timdamen** [comment]
The vite example is building and running fine. Only I'm not sure how to instead of console logging the remote components, display them on the screen. Could you provide an example of it?

**Comment by gioboa** [comment]
Thanks @zhangHongEn for your contribution.

**Comment by zhangHongEn** [comment]
> Looks great @zhangHongEn, the vite example is building and running fine. Only I'm not sure how to instead of console logging the remote components, display them on the screen. Could you provide an example of it?

Ok, I'll add it

**Comment by gioboa** [comment]
That's a great piece of code 👏
I'm wondering if we can include `module-federation-vite` code inside the repository rather than using the dependency (see image below)

**Comment by zhangHongEn** [comment]
@gioboa https://github.com/module-federation/vite/pull/28/commits/e5e37ece2b5b25f95105527b84ef93215cd7dbc9

File: packages/examples-rust/dist/@mf-types.d.ts
Comment (husayt) [inline packages/examples-rust/dist/@mf-types.d.ts:1]: @zhangHongEn do we need to include dist folder here?

Comment (zhangHongEn) [reply packages/examples-rust/dist/@mf-types.d.ts:1]: Demo output, no effect

**Comment by husayt** [comment]
@gioboa would still be good to include your original native-federation wrapper under either `/module-federation/vite/packages/native-federation` or `/module-federation/native-federation`  once this merged. This repo will have best of both worlds literally then: Module Federation for vite and Native Federation for Vite.

Thank you

File: README.md [hunk 1]
```diff
@@ -10,55 +10,75 @@ This plugin makes Module Federation work together with [Vite](https://vitejs.dev
 ## Working implementations
 
 ### [React](https://github.com/module-federation/module-federation-examples/tree/master/vite-react-microfrontends)<br>
+
```
Comment (husayt) [inline README.md:13]: @zhangHongEn are these valid examples still? as they don't use this libary

File: package.json [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-    "build": "rimraf lib && microbundle --no-sourcemap --compress=false"
+    "dev:examples-vite": "cd packages/examples-vite && npm run dev",
+    "dev:examples-rust": "cd packages/examples-rust && npm run dev",
+    "dev": "npx -y npm-run-all --parallel dev:examples-vite dev:examples-rust",
+    "build:examples-vite": "cd packages/examples-vite && npm run build",
+    "build:examples-rust": "cd packages/examples-rust && npm run build",
+    "build": "npx -y npm-run-all --parallel build:examples-vite build:examples-rust",
+    "preview": "cd packages/examples-vite && npm run preview"
+  },
+  "author": {
```
Comment (husayt) [inline package.json]: @zhangHongEn at least you should have your name under

"contributors": []

This is great effort and needs to be recognised. Thank you

**Comment by gioboa** [comment]
@zhangHongEn thanks for rocking with this PR. I'm wondering if we can use 
'@module-federation/esbuild' code in this implementation? #27 tried to implement the functionality but it's abandoned I guess

**Comment by zhangHongEn** [comment]
> @zhangHongEn thanks for rocking with this PR. I'm wondering if we can use '@module-federation/esbuild' code in this implementation? #27 tried to implement the functionality but it's abandoned I guess

Vite's temporary dev mode uses esbuild. I have tried configuring @module-federation/esbuild in vite.config.js optimizeDeps.esbuildOptions.plugins, but it seems to conflict with vite's pre-built

**Comment by husayt** [comment]
> @zhangHongEn thanks for rocking with this PR. I'm wondering if we can use '@module-federation/esbuild' code in this implementation? #27 tried to implement the functionality but it's abandoned I guess

While it might be useful to look into `https://www.npmjs.com/package/@module-federation/esbuild?activeTab=code` this for reference, I would not recommend using that library as Vite wraps esbuild/rollup and tries to provide unified interface, so best approach to use vite holistically.

**Comment by zhangHongEn** [comment]
> > @zhangHongEn thanks for rocking with this PR. I'm wondering if we can use '@module-federation/esbuild' code in this implementation? #27 tried to implement the functionality but it's abandoned I guess
> 
> While it might be useful to look into `https://www.npmjs.com/package/@module-federation/esbuild?activeTab=code` this for reference, I would not recommend using that library as Vite wraps esbuild/rollup and tries to provide unified interface, so best approach to use vite holistically.

Yes, the vite plugin and esbuild plugin are independent of each other

**Comment by husayt** [comment]
@gioboa is there anything else blocking this PR's merge. Will be great to get this out to keep momentum on.

**Review by husayt — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (zhangHongEn) [reply README.md:13]: Invalid, forgot to delete

**Comment by gioboa** [comment]
@husayt this is a good POC, but I can't release a new version of the library with this code. 
I'll try to adapt this code to and cut a new release.
I also need to convert most of the files to TypeScript because right now are in pure JavaScript.

File: package.json (hunk 2 above)
Comment (zhangHongEn) [reply package.json]: Thank you for your recognition

**Comment by zhangHongEn** [comment]
> @husayt this is a good POC, but I can't release a new version of the library with this code. I'll try to adapt this code to and cut a new release. I also need to convert most of the files to TypeScript because right now are in pure JavaScript.
> 
> I tried to use this code and the `module-federation-vite` package too in this repo [vue-microfrontend-demo](https://github.com/gioboa/vue-microfrontend-demo) but I can't get it to work. @zhangHongEn Can you open a PR in that repo to show me how replace the old code with the new one pls?

It's a good test scenario indeed.
https://github.com/gioboa/vue-microfrontend-demo/pull/8

**Comment by gioboa** [comment]
> https://github.com/gioboa/vue-microfrontend-demo/pull/8

Thanks 🚀

**Comment by zhangHongEn** [comment]
@gioboa Please create a new branch and merge the code first. I received some issues and PRs.
https://github.com/zhangHongEn/module-federation-vite/pull/4

**Comment by gioboa** [comment]
> > [gioboa/vue-microfrontend-demo#8](https://github.com/gioboa/vue-microfrontend-demo/pull/8)
> 
> Thanks 🚀

Thanks for your commitment.
I tested your PR in the Vue mfe example and it's not working, I have multiple errors during the `build` process.
I would like to have a fully working application here -> https://github.com/gioboa/vue-microfrontend-demo/pull/8
before going on with this new implementation.

**Comment by zhangHongEn** [comment]
> > > [gioboa/vue-microfrontend-demo#8](https://github.com/gioboa/vue-microfrontend-demo/pull/8)
> > 
> > 
> > Thanks 🚀
> 
> Thanks for your commitment. I tested your PR in the Vue mfe example and it's not working, I have multiple errors during the `build` process. I would like to have a fully working application here -> [gioboa/vue-microfrontend-demo#8](https://github.com/gioboa/vue-microfrontend-demo/pull/8) before going on with this new implementation.

Done, this is not a plugin issue, but I will fix it in the plugin
``` js
 manualChunks(id) {
  if (id.indexOf("/@module-federation/runtime") > -1) {
    return "mfruntime";
  }
}
```

**Comment by zhangHongEn** [comment]
> @gioboa Please create a new branch and merge the code first. I have received some issues and PRs and need to synchronize the code. Please give me permission so that I can quickly merge in the new non-trunk branch. Before the complete merge, I will publish the npm module-federation-vite package for use
> 
> [zhangHongEn/module-federation-vite#4](https://github.com/zhangHongEn/module-federation-vite/pull/4)

@gioboa  Can we create a new non-main branch and work together first?

**Comment by gioboa** [comment]
I created a `new-config` branch

**Review by gioboa — APPROVED** [review]

**Comment by husayt** [comment]
> I created a `new-config` branch

@zhangHongEn you can create a new PR for follow up changes.

But here are some fixes which would save you time

 in `pnpm-workspace.yaml`

```
packages:
  - 'examples/*'
  - '.'
```

Replace 'module-federation-vite' to '@module-federation/vite' everywhere

Replace `import path from 'path';` to  ` import {join} from 'path';`

remove `export default federation;` as only named export should be there

**Comment by zhangHongEn** [comment]
@husayt #29

**Comment by luokuning** [comment]
@zhangHongEn Awesome! Do we need to publish a new version and update doc to provide the latest configuration?

**Comment by zhangHongEn** [comment]
> @zhangHongEn Awesome! Do we need to publish a new version and update doc to provide the latest configuration?
I will solve these two important problems tonight. #66 #40 
Currently still publishing @module-federation/vite@next to npm

**Comment by luokuning** [comment]
@zhangHongEn Thank you for your effort! Looking forward to the release of version 1.0.

**Pull request merged**